### PR TITLE
Add a to_table() method to TimeMapAxis

### PR DESCRIPTION
**Pull request by cgalelli**
This simple PR adds a `to_table` method for TimeMapAxis that produces the same table as the one produced by the equivalent GTIs.
Mentioned by @AtreyeeS as a simple possibility to spare a couple of function calls in PR #5443 

Probably needs to be completed with some format and validity checks.

### CODE REVIEW

File: gammapy/maps/axes.py [hunk 1]
```diff
@@ -3030,6 +3030,25 @@ def to_gti(self):
...
+
+        Parameters
+        ----------
+        format : {"gadf"}
+            Format specification. Default is "gadf".
+
+        Returns
+        -------
+        table : `~astropy.table.Table`
+            Table with axis data.
```
Comment (bkhelifi) [inline gammapy/maps/axes.py:3044]: The table contains astopy.Time object only. Is not it?

File: gammapy/maps/axes.py [hunk 2]
```diff
@@ -3030,6 +3030,25 @@ def to_gti(self):
             self.edges_min, self.edges_max, reference_time=self.reference_time
         )
 
+    def to_table(self, format="gadf"):
+        """Create table.
+
+        Parameters
+        ----------
+        format : {"gadf"}
+            Format specification. Default is "gadf".
```
Comment (registerrier) [inline gammapy/maps/axes.py:3039]: Is there a specification for such a table in GADF? Usually time quantities are stored with a reference value stored as MJDREFF in the header and start and stops are just time like quantities.

**Review by registerrier — COMMENTED** [review]
Thanks @cgalelli . In general, it is good to uniformize the API between regular `MapAxis` and `TimeMapAxis` as the latter should be subclasses of the former.

Practically, it is not clear to me that serialization method should live on basic building blocks. But that's probably another question.

File: gammapy/maps/axes.py (hunk 2 above)
Comment (AtreyeeS) [inline gammapy/maps/axes.py:3039]: We can also use `self.to_gti().to_table()` to ensure a similar serialisation

File: gammapy/maps/axes.py (hunk 1 above)
Comment (cgalelli) [reply gammapy/maps/axes.py:3044]: Yes the table would contain only times, at least in the basic format

File: gammapy/maps/axes.py (hunk 2 above)
Comment (cgalelli) [reply gammapy/maps/axes.py:3039]: You are right, the specification would be to give the time as a quantity, like doing `to_gti().table`. That is probably the best solution.

**Review by AtreyeeS — COMMENTED** [review]
Thanks @cgalelli ! Can you add a test please

**Comment by cgalelli** [comment]
Thanks, @AtreyeeS , tests are also missing for `MapAxis.to_table()`. Should they be added as well? And how should they look? Maybe I can make a dedicated PR

**Comment by AtreyeeS** [comment]
>Tests are also missing for `MapAxis.to_table()`. Should they be added as well? And how should they look? Maybe I can make a dedicated PR

Yes should definitely be added! I think you can simply assert that the table column name matches the one of the MapAxis, and that the table values correspond to the MapAxis nodes

File: gammapy/maps/axes.py [hunk 3]
```diff
@@ -3030,6 +3030,23 @@ def to_gti(self):
             self.edges_min, self.edges_max, reference_time=self.reference_time
         )
 
+    def to_table(self, format="gadf"):
+        """Create table.
```
Comment (AtreyeeS) [inline gammapy/maps/axes.py:3034]: The format argument is unused

**Review by AtreyeeS — COMMENTED** [review]
Please remove the  `format` argument. Good from my side

File: gammapy/maps/axes.py (hunk 1 above)
Comment (bkhelifi) [reply gammapy/maps/axes.py:3039]: Should we precise it explicitly? An open question...

**Review by bkhelifi — APPROVED** [review]
Thanks a lot!

**Review by AtreyeeS — APPROVED** [review]
Thanks @cgalelli ! Merging

**Pull request merged**